spec/observables/defer-spec.ts

```diff
@@ -72,4 +72,16 @@ describe('Observable.defer', () => {
     expectObservable(e1, unsub).toBe(expected);
     expectSubscriptions(source.subscriptions).toBe(sourceSubs);
   });
+
+  type(() => {
```

Comment:
I don't think there's wrong time, or even it'd be great time since we don't have much test for this yet. :)

Basically this test is ensuring type definition is 'works' as it's actual functionality does, for instance in case of this one `defer`, type should support cases

- when factory function returns `observable<T>`, subscription should carry over type T so you can access its properties without type errors
- factory function can throw without return anything, so defer should accept function throws but does not return

both of declaration in line 82 / 83 illustrates those usecases, and since it's compile time it is being verified when test case is being compiled. If those are not working as intended, compiler will throw out messages of expected type vs. provided one.

I also agree it could be possibly confusing to consume this kind of test (small one would be ok, but case like this? https://github.com/ReactiveX/RxJS/pull/1189/files) and it'd be great if there's suggestion how to improve this or requirement should be added for better visibility. Since there is barely no test yet, changes can be easily introduced.